enterprise/server/backends/distributed/distributed.go

```diff
@@ -19,26 +20,46 @@ import (
 )
 
 type Cache struct {
-	local     interfaces.Cache
-	myAddr    string
-	groupName string
-	prefix    string
+	local             interfaces.Cache
+	myAddr            string
+	groupName         string
+	prefix            string
+	replicationFactor int
 
 	cacheProxy       *cacheproxy.CacheProxy
 	consistentHash   *consistent_hash.ConsistentHash
 	heartbeatChannel *heartbeat.HeartbeatChannel
 }
 
-func NewDistributedCache(env environment.Env, c interfaces.Cache, myAddr, groupName string) (*Cache, error) {
+func NewDistributedCache(env environment.Env, c interfaces.Cache, myAddr, groupName string, replicationFactor int) (*Cache, error) {
 	chash := consistent_hash.NewConsistentHash()
-	return &Cache{
-		local:            c,
-		cacheProxy:       cacheproxy.NewCacheProxy(env, c, myAddr),
-		myAddr:           myAddr,
-		groupName:        groupName,
-		consistentHash:   chash,
-		heartbeatChannel: heartbeat.NewHeartbeatChannel(env.GetPubSub(), myAddr, groupName, chash.Set),
-	}, nil
+	dc := &Cache{
+		local:             c,
+		cacheProxy:        cacheproxy.NewCacheProxy(env, c, myAddr),
+		myAddr:            myAddr,
+		groupName:         groupName,
+		consistentHash:    chash,
+		heartbeatChannel:  heartbeat.NewHeartbeatChannel(env.GetPubSub(), myAddr, groupName, chash.Set),
+		replicationFactor: replicationFactor,
+	}
+	go func() {
+		dc.StartListening()
+	}()
+	return dc, nil
+}
+
+func (c *Cache) StartListening() {
```

Comment:
It's a good idea but feels a little overkill for now because the lifecycle of these should match the server: basically you start one and run it until server is killed.

(I really only added the StartListening/Shutdown stuff so that tests could bring up and down a bunch of these quickly...)